builtin/credential/approle/path_role.go

```diff
@@ -2358,12 +2359,27 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 		return nil, errwrap.Wrapf("failed to store secret_id: {{err}}", err)
 	}
 
-	return &logical.Response{
+	resp := &logical.Response{
 		Data: map[string]interface{}{
 			"secret_id":          secretID,
 			"secret_id_accessor": secretIDStorage.SecretIDAccessor,
 		},
-	}, nil
+	}
+
+	// Follow the behavior from registerSecretIDEntry and set the response TTL
+	// to the entry's TTL unless it's negative or greater than the system's max
+	// lease TTL.
+	switch {
+	case secretIDStorage.SecretIDTTL < time.Duration(0):
+		// This case should not be possible since TypeDurationSecond does not
+		// support negative values, but we follow the same convention as
+		// registerSecretIDEntry
+		resp.Data["ttl"] = int(b.System().MaxLeaseTTL().Seconds())
```

Comment:
Do we want to use `secret_id_ttl` instead? My thoughts are that since this isn't a standard lease, and we're just relaying configuration details, it might be better to be different. Also, since we're going to look for these details in consul-template, it could reduce the chance of unintentionally picking up some other secret as AppRole.

Reply:
Done in 8b90861.